types/src/share.rs

```diff
@@ -134,16 +145,44 @@ impl Share {
         }
     }
 
+    /// Get the `signer` part of share if it's first in the sequence of sparse shares and `share_version` supprots
+    /// it. Otherwise returns `None`
+    pub fn signer(&self) -> Option<AccAddress> {
+        let info_byte = self.info_byte()?;
+        if info_byte.is_sequence_start()
+            && info_byte.version() == appconsts::SHARE_VERSION_ONE
+            && !self.namespace().is_reserved()
```

Comment:
I prefer something like `!self.is_compact()` or a comment in both cases where is_reserved is used. It tooks me a bit of time to cross check this line with the specification